### ISSUE TITLE
One time batch jobs and preparation SQL scripts

### ISSUE THREAD

**Issue by kabirbaidhya** [leapfrogtechnology/sync-db#17]
## Feature Discussion
**Batch jobs with preparation scripts**

### Problem background
One common use case is to run one-time batch job or data operation - which is a series of SQL scripts to be run on N databases. 

**Consider these examples:**
 - ETL job running a series of data transformation logic (procedure calls) on source tables to prepare destination tables with processed data
 - data analysis - series of data processing (procedure calls) to analyse the data from tables and write analysed outputs to destination tables. 
 - or any kind of batch data operations job

In these kinds of jobs, the logical components are not needed in the DB once the job is complete so they could be dropped. 

### Proposed Implementation
Example of the config of how we can achieve this. 
```
sql:
  prep:
    - schema/transformation.sql
    - view/transformation/view1.sql
    - view/transformation/view1.sql
    - function/transformation/some_func.sql
    - procedure/transformation/proc1.sql
    - procedure/transformation/proc2.sql

hooks:
  script:
    - scripts/run_transform.sql
```
**Steps**
 - drop (if exists) the objects under `sql.prep` in reverse order
 - create the objects under `sql.prep` in the normal order
 - run the sql scripts `hooks.script`
 - drop (if exists) the objects under `sql.prep` in reverse order

This isn't a database synchronization operation but a general batch job running mechanism. 

**TLDR - no db objects exists in the db once the batch job (script) is executed.**

@mesaugat @SafalPandey Let me know what do you think?
